src/main/java/traincraft/blocks/trainworkbench/TileTrainWorkbench.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ *
+ * This file ("TileTrainWorkbench.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ *
+ * @author PseudonymPatel
+ * @since 2020-7-30
+ */
+
+package traincraft.blocks.trainworkbench;
+
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import traincraft.api.InventorySpecific;
+import traincraft.tile.BaseTile;
+
+import javax.annotation.Nonnull;
+
+//TODO: make the trainworkbench save inventory on close
+//TODO: BUG: closing game while container is open will delete everything inside of container.
+public class TileTrainWorkbench extends BaseTile {
```

Comment:
please drop the contents when closed, as long as they don't stay in them

Reply:
That's what it does currrently, like the vanilla crafting table.

Reply:
ok